ethapi/api.go

```diff
@@ -1547,7 +1547,7 @@ func (s *PublicTransactionPoolAPI) GetTransactionByHash(ctx context.Context, has
 	}
 	if tx != nil {
 		header, err := s.b.HeaderByNumber(ctx, rpc.BlockNumber(blockNumber))
-		if err != nil {
+		if header == nil || err != nil {
```

Comment:
Basically it's okay to check this. 

Or it'd better to do it in other way by letting `HeaderByNumber()` will return `errors.New("xxx not found")` then we don't need to check **_nil_** anymore?

Similar to `BlockByNumber()` and other API methods.

Reply:
Current behavior (returning `nil, nil` from `HeaderByNumber`) is consistent with go-ethereum, for example:

https://github.com/ethereum/go-ethereum/blob/master/eth/api_backend.go#L75

If we change it, it can bring compatibility issues in porting code from go-ethereum.

Reply:
okay, I got it.

Actually, go-ethereum is not consistent as well, `HeaderByNumberOrHash` may return `return nil, errors.New("header for hash not found")` as in 

https://github.com/ethereum/go-ethereum/blob/862f8e98bce89889de22e2d32bf37f823127a32c/eth/api_backend.go#L85